src/types/Health.ts

```diff
@@ -151,9 +151,7 @@ export const getRequestErrorsViolations = (reqIn: ThresholdStatus, reqOut: Thres
 };
 
 export abstract class Health {
-  items: HealthItem[];
-
-  constructor(items: HealthItem[]) {
+  constructor(public items: HealthItem[], public requests: RequestHealth, public ctx: HealthContext) {
```

Comment:
I will look at this PR more closely tomorrow, but why do we need to keep `requests` and `ctx` as class members? They should only be necessary while building the objects, and be discarded after, no?

Reply:
Yes, I also understand that requests and ctx don't need to be class fields and that data can be discarded.

Looking at the child classes, these fields are public. So, this is only pulling up the existent definitions. But, definitely, those class fields can probably can be removed.

Reply:
True, I changed the constructors to keep the interface the same across subclasses and have a smaller diff without too many architectural changes, because doing it the other way (remove the args from the subclasses) breaks the public interface.